src/Workspaces/Core/Portable/SymbolId/SymbolKey.Visitor.cs

```diff
@@ -71,9 +67,17 @@ public override SymbolKey VisitModule(IModuleSymbol moduleSymbol)
 
             public override SymbolKey VisitNamedType(INamedTypeSymbol namedTypeSymbol)
             {
-                return namedTypeSymbol.TypeKind == TypeKind.Error
-                    ? new ErrorTypeSymbolKey(namedTypeSymbol, this)
-                    : (SymbolKey)new NamedTypeSymbolKey(namedTypeSymbol, this);
+                if (namedTypeSymbol.TypeKind == TypeKind.Error)
+                {
+                    return new ErrorTypeSymbolKey(namedTypeSymbol, this);
+                }
+
+                if (namedTypeSymbol.IsTupleType)
+                {
+                    return new TupleTypeSymbolKey((ITupleTypeSymbol)namedTypeSymbol, this);
```

Comment:
ITupleTypeSymbol didn't derive from NamedTypeSymbol.  So it's super weird that this works.

Also, shouldn't we have VisitTupleType, instead of VisitNamedType?